lib/arborist/reify.js

```diff
@@ -233,7 +234,7 @@ module.exports = cls => class Reifier extends cls {
     const actualOpt = this[_global] ? {
       ignoreMissing: true,
       global: true,
-      filter: (node, kid) => !node.isRoot && node !== node.root.target
+      filter: (node, kid) => this[_updateAll] || (!node.isRoot && node !== node.root.target)
```

Comment:
If it's global, could we just check for `this[_explicitRequests].size === 0`?

Reply:
Would save having to add another shared symbol.

Reply:
yeah that would probably be fine, i went with the shared symbol just because it's more clear what we're actually asserting

Reply:
That second conditional can also be shortened to `!node.isProjectRoot` now.

Reply:
:+1: updated that conditional to not use the shared symbol and use isProjectRoot